### PR TITLE
ICS30: Update callbacks

### PR DESCRIPTION
**Pull request by AdityaSripal**
Update based on #629

### CODE REVIEW

File: spec/app/ics-030-middleware/README.md [hunk 1]
```diff
@@ -101,6 +108,12 @@ function onChanOpenInit(
         counterpartyChannelIdentifier,
         appVersion, // note we only pass app version here
     )
+    abortTransactionUnless(err != nil)
```
Comment (mpoke) [inline spec/app/ics-030-middleware/README.md:111]: If we call `abortTransactionUnless` here, is there a reason to return `Error`?

File: spec/app/ics-030-middleware/README.md [hunk 2]
```diff
@@ -130,6 +144,12 @@ function OnChanOpenTry(
           cpAppVersion, // note we only pass counterparty app version here
           appVersion, // only pass app version
       )
+      abortTransactionUnless(err != nil)
```
Comment (mpoke) [inline spec/app/ics-030-middleware/README.md:147]: ditto

**Review by mpoke — APPROVED** [review]
See my comment below.

File: spec/app/ics-030-middleware/README.md (hunk 1 above)
Comment (AdityaSripal) [reply spec/app/ics-030-middleware/README.md:111]: This is true, but I'm just trying to stay consistent with the ICS26 function signature. In practice, the abortTransaction process is carried out by returning an error at this level.

**Pull request merged**